src/util/projection.ts

```diff
@@ -0,0 +1,51 @@
+import {PrimitiveProjection, ProjectionTransition, ProjectionType, PropertyValueSpecification} from '../types.g';
+
+export class Projection {
+    readonly from: PrimitiveProjection;
+    readonly to: PrimitiveProjection;
+    readonly transition: number;
+    constructor(from: PrimitiveProjection, to: PrimitiveProjection, transition: number){
+        this.from = from;
+        this.to = to;
+        this.transition = transition;
+    }
+
+    static parse(input: Projection | string | undefined | null): Projection | undefined {
```

Comment:
Is this method being used somewhere in the code? I haven't seen it, I'm guessing it is important, right?